ethereum/rpc/namespaces/miner/api.go

```diff
@@ -0,0 +1,163 @@
+package miner
+
+import (
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/config"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tharsis/ethermint/ethereum/rpc/namespaces/eth"
+	rpctypes "github.com/tharsis/ethermint/ethereum/rpc/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+// API is the miner prefixed set of APIs in the Miner JSON-RPC spec.
+type API struct {
+	ctx    *server.Context
+	logger log.Logger
+	ethAPI *eth.PublicAPI
+}
+
+// NewMinerAPI creates an instance of the Miner API.
+func NewMinerAPI(
+	ctx *server.Context,
+	ethAPI *eth.PublicAPI,
+) *API {
+	return &API{
+		ctx:    ctx,
+		ethAPI: ethAPI,
+		logger: ctx.Logger.With("module", "miner"),
+	}
+}
+
+// SetEtherbase sets the etherbase of the miner
+func (api *API) SetEtherbase(etherbase common.Address) bool {
+	api.logger.Debug("miner_setEtherbase")
+
+	list, err := api.ethAPI.ClientCtx().Keyring.List()
+	if err != nil || len(list) < 1 {
+		api.logger.Debug("Could not get list of addresses")
+		return false
+	}
+
+	addr := common.BytesToAddress(list[0].GetPubKey().Address())
```

Comment:
the delegator address should be the one from the eth API `Coinbase()`. You can probably move that function to the `Backend` so that `SetEtherbase` and `Coinbase` can use it:

```go
func (e EVMBackend) GetCoinbase() (sdk.AccAddress, error) {
	node, err := e.clientCtx.GetNode()
	if err != nil {
		return "", err
	}

	status, err := node.Status(e.ctx)
	if err != nil {
		return "", err
	}

	req := &evmtypes.QueryValidatorAccountRequest{
		ConsAddress: sdk.ConsAddress(status.ValidatorInfo.Address).String(),
	}

	res, err := e.queryClient.ValidatorAccount(e.ctx, req)
	if err != nil {
		return "", err
	}

	address, _ := sdk.AccAddressFromBech32(res.AccountAddress)
	return address, nil
}
```